docs/api/primitives.md

```diff
@@ -22,39 +21,62 @@ between date and time part, ”hh” denotes hour in the 24-hour format (00-23),
 ”2024-12-05T12:00:00Z”
 
 ## Direction
-The direction of a sorting operation. Possible values are the representation of ascending or descending order.
+
+The direction of a sorting operation. Possible values are the String representation of ascending (`ASC`) or descending (`DESC`) order.
 
 ## ErrorType
 
 String value of the error type. Could be `ARROWHEAD`, `INVALID_PARAMETER`, `AUTH`, `FORBIDDEN`, `DATA_NOT_FOUND`,
 `TIMEOUT`, `LOCKED`, `INTERNAL_SERVER_ERROR` or `EXTERNAL_SERVER_ERROR`.
 
-## InterfaceTemplate
-A string identifier of an interface descriptor.
-
 ## MQTTQoS
 
 QoS in MQTT refers to the level of guarantee for message delivery between the publisher and the subscriber. It can be `0`, `1` or `2` ([learn more](https://www.hivemq.com/blog/mqtt-essentials-part-6-mqtt-quality-of-service-levels/)).
 
 ## Name
 
-A String indentifier that is intended to be both human and machine-readable.
+A String indentifier that is intended to be both human and machine-readable. The allowed characters are letters, numbers and dash (`-`). A name has to start with a letter and cannot end with a dash. There are different restrictions on the length of the name depending on the data model.
 
 ## Number
 
 Decimal number.
 
 ## PropertyValidator
-An identifier of any suitable validator function chosen by the implementor of service.
 
+An identifier of any suitable validator function chosen by the implementor of service. The validators have different kinds of inputs, depending on what to validate. Some validators have optional or mandatory **arguments**. The implemented validators are the following:
+- `NOT_EMPTY_ADDRESS_LIST`: The **input** is a list of [Addresses](#address) and must contain at least one element. The validator checks if the addresses are valid and in alignment with the supported address types and performs normalization. An example for usage: validating HTTP interface access addresses.
+- `NOT_EMPTY_STRING_SET`: The **input** is a list of [Strings](#string) and must contain at least one element. The validator checks if the are not empty and performs normalization. It can be used with the optional `NAME` **argument**. In that case, normalization and validation happens according to the naming convention. An example for usage: validating possible MQTT inerface operations.
+- `PORT`: This validator is meant to be used for ports. The **iput** is an integer that represents a port number. The validator checks if the given value is between 1 and 65535, since in practice all valid port numbers fall within this range.
+- `MINMAX`: The **input** is a [Number](#number). It is mandatory to give two [Strings](#string) as **arguments**, that represent the lower and upper limits of a closed interval. The validator checks if the input falls within the interval. (The `PORT` validator is specific `MINMAX` validator, where the arguments are 1 and 65535.)
```

Comment:
is specific => is a specific
where the arguments are 1 and 65535 => where the arguments are "1" and "65535"